src/effects/native/gatereffect.h

```diff
@@ -0,0 +1,51 @@
+#ifndef GATEREFFECT_H
+#define GATEREFFECT_H
+
+#include "effects/effectprocessor.h"
+#include "engine/effects/engineeffect.h"
+#include "engine/effects/engineeffectparameter.h"
+#include "util/class.h"
+#include "util/defs.h"
+#include "util/sample.h"
+#include "util/types.h"
+
+
+struct GaterGroupState {
+    enum State {IDLE, ATTACK, HOLD, RELEASE} state;
```

Comment:
Use C++ "enum class" to improve type safety and limit visibility. The latter is not an issue here, but might be elsewhere.